app/views/layouts/application.html.erb

```diff
@@ -10,10 +10,14 @@
   </head>
 
   <body>
-     <%= render 'layouts/header' %>
+    <%= render 'layouts/header' %>
     <div class="container">
+      <% flash.each do |message_type, message| %>
```

Comment:
nguyên lý hoạt động của `flash` là gì? so sánh flash vs flash.now

Reply:
flash sẽ in ra thông báo sau khi có một request được thực hiện, flash sẽ tự động xuất hiện trong khi flash.now sẽ chỉ xuất hiện trên trang web hiện tại và sẽ biến mất ngay sau khi có một request mới được thêm vào

Reply:
@dungvnk1
 flash làm được như vậy dữa trên công nghệ hay kĩ thuật gì?

Reply:
theo em tìm hiểu thì flash là một middleware method, nó sẽ được lưu vào trong session và có thể được access thông qua những requests